lang/cs/Org.Apache.REEF.Client/Multi/MultiRuntimeConfigurationBuilder.cs

```diff
@@ -0,0 +1,256 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Org.Apache.REEF.Client.Multi.Parameters;
+using Org.Apache.REEF.Client.Yarn;
+using Org.Apache.REEF.Client.YARN;
+using Org.Apache.REEF.Client.YARN.HDI;
+using Org.Apache.REEF.Client.YARN.Parameters;
+using Org.Apache.REEF.Common.Runtime;
+using Org.Apache.REEF.IO.FileSystem.AzureBlob;
+using Org.Apache.REEF.Tang.Formats;
+using Org.Apache.REEF.Tang.Implementations.Configuration;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Collections;
+
+namespace Org.Apache.REEF.Client.Multi
+{
+    /// <summary>
+    /// Builder for a multi runtime configuration.
+    /// Generates a configuration that instantiate a ReefClient that submits multi runtime job
+    /// </summary>
+    public sealed class MultiRuntimeConfigurationBuilder
+    {
+        private const string MultiRuntimeLauncher =
+            "org.apache.reef.bridge.client.MultiRuntimeYarnBootstrapREEFLauncher";
+
+        private static readonly ReadOnlySet<RuntimeName> SupportedRuntimes =
+            new ReadOnlySet<RuntimeName>(new[] { RuntimeName.Yarn, RuntimeName.Local });
+
+        private HashSet<RuntimeName> _runtimes = new HashSet<RuntimeName>();
+
+        private Optional<RuntimeName?> _defaultRuntime = Optional<RuntimeName?>.Empty();
```

Comment:
For some reason when Optional is used with enum it always have value (default value for enum).